### PR TITLE
Better error for failed lookup

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/2965

### CODE REVIEW

**Review by axic — CHANGES_REQUESTED** [review]
Please also include the library specific test the original issue had.

File: test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+library L {
+  function f(uint, uint) {}
+  function f(uint) {}
+}
+contract C {
+  function g() { L.f(1, 2, 3); }
+}
+// ----
+// TypeError: (94-97): Member "f" not found or not visible after argument-dependent lookup in type(library L)
```
Comment (axic) [inline test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol:9]: Added this test case from the original issue.

**Review by axic — APPROVED** [review]

File: test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol (hunk 1 above)
Comment (axic) [reply test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol:9]: A better question though: cannot we support overloads from libraries?

Comment (chriseth) [reply test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol:9]: Don't we?

Comment (axic) [reply test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol]: The above example is broken and doesn't fullfill the actual test purposes it should.

There is no overload with 3 arguments. The call should be changed to two.

Comment (axic) [reply test/libsolidity/syntaxTests/memberLookup/failed_function_lookup_in_library.sol:9]: Never mind, it was too late.

**Pull request merged**